source/common/config/well_known_names.h

```diff
@@ -0,0 +1,174 @@
+  // Echo filter
+  const std::string ECHO = "envoy.echo";
+  // HTTP connection manager filter
+  const std::string HTTP_CONNECTION_MANAGER = "envoy.http_connection_manager";
+  // Mongo proxy filter
+  const std::string MONGO_PROXY = "envoy.mongo_proxy";
+  // Rate limit filter
+  const std::string RATE_LIMIT = "envoy.ratelimit";
+  // Redis proxy filter
+  const std::string REDIS_PROXY = "envoy.redis_proxy";
+  // IP tagging filter
+  const std::string TCP_PROXY = "envoy.tcp_proxy";
+
+  // Converts names from v1 to v2
+  const V1Converter v1_converter_;
+
+  NetworkFilterNameValues()
+      : v1_converter_({CLIENT_SSL_AUTH, ECHO, HTTP_CONNECTION_MANAGER, MONGO_PROXY, RATE_LIMIT,
+                       REDIS_PROXY, TCP_PROXY}) {}
+};
+
+typedef ConstSingleton<NetworkFilterNameValues> NetworkFilterNames;
+
+/**
+ * Well-known http filter names.
+ */
+class HttpFilterNameValues {
+public:
+  // Buffer filter
+  const std::string BUFFER = "envoy.buffer";
+  // Dynamo filter
+  const std::string DYNAMO = "envoy.http_dynamo_filter";
+  // Fault filter
+  const std::string FAULT = "envoy.fault";
+  // GRPC http1 bridge filter
+  const std::string GRPC_HTTP1_BRIDGE = "envoy.grpc_http1_bridge";
+  // GRPC json transcoder filter
+  const std::string GRPC_JSON_TRANSCODER = "envoy.grpc_json_transcoder";
+  // GRPC web filter
+  const std::string GRPC_WEB = "envoy.grpc_web";
+  // IP tagging filter
+  const std::string IP_TAGGING = "envoy.ip_tagging";
+  // Rate limit filter
+  const std::string RATE_LIMIT = "envoy.rate_limit";
+  // Router filter
+  const std::string ROUTER = "envoy.router";
+
+  // Converts names from v1 to v2
+  const V1Converter v1_converter_;
+
+  HttpFilterNameValues()
+      : v1_converter_({BUFFER, DYNAMO, FAULT, GRPC_HTTP1_BRIDGE, GRPC_JSON_TRANSCODER, GRPC_WEB,
+                       IP_TAGGING, RATE_LIMIT, ROUTER}) {}
+};
+
+typedef ConstSingleton<HttpFilterNameValues> HttpFilterNames;
+
+/**
+ * Well-known access log names.
+ */
+class HttpTracerNameValues {
+public:
+  // Lightstep tracer
+  const std::string LIGHTSTEP = "envoy.lightstep";
+  // Zipkin tracer
+  const std::string ZIPKIN = "envoy.zipkin";
+};
+
+typedef ConstSingleton<HttpTracerNameValues> HttpTracerNames;
+
+/**
+ * Well-known stats sink names.
+ */
+class StatsSinkNameValues {
+public:
+  // Statsd sink
+  const std::string STATSD = "envoy.statsd";
+};
+
+typedef ConstSingleton<StatsSinkNameValues> StatsSinkNames;
+
+/**
+ * Well-known access log names.
+ */
+class AccessLogNameValues {
+public:
+  // File access log
+  const std::string FILE = "envoy.file_access_log";
+};
+
+typedef ConstSingleton<AccessLogNameValues> AccessLogNames;
+
+/**
+ * Well-known metadata filter namespaces.
+ */
+class MetadataFilterValues {
+public:
+  // Filter namespace for built-in load balancer.
+  const std::string ENVOY_LB = "envoy.lb";
+  // Filter namespace for built-in router opaque data.
+  const std::string ENVOY_ROUTER = "envoy.router";
```

Comment:
I think ENVOY_ROUTER is a dupe of ROUTER above. The idea in the metadata namespace is you use the filter name as key.

Reply:
Does this mean that these should all just be treated as filter names?